workflow-manager/main.go

```diff
@@ -483,14 +483,36 @@ func fmtTime(t time.Time) string {
 	return t.Format("2006/01/02/15/04")
 }
 
-// idForJobName generates a job name-safe string from an aggregationID.
+// jobNameForBatchPath generates a name for the Kubernetes job that will intake
+// the provided batch. The name will incorporate the aggregation ID, batch UUID
+// and batch timestamp while being a legal Kubernetes job name.
+func intakeJobNameForBatchPath(path *batchPath) string {
+	// Kubernetes job names must be valid DNS identifiers, which means they are
+	// limited to 63 characters in length and also what characters they may
+	// contain. Intake job names are like:
+	// i-<aggregation name fragment>-<batch UUID fragment>-<batch timestamp>
+	// The batch timestamp is 16 characters, and the 'i' and '-'es take up
+	// another 4, leaving 43. We take the '-'es out of the UUID and use half of
+	// it, hoping that this plus the date will provide enough entropy to avoid
+	// collisions. Half a UUID is 16 characters, leaving 27 for the aggregation
+	// ID fragment.
+	// For example, we might get:
+	// i-com-apple-EN-verylongnameth-0f0f0f0f0f0f0f0f-2006-01-02-15-04
+	return fmt.Sprintf("i-%s-%s-%s",
+		aggregationJobNameFragment(path.aggregationID, 27),
+		strings.ReplaceAll(path.ID, "-", "")[:16],
```

Comment:
For a tiny bit more entropy, we could use the last 16 chars of the UUID which is "more random" than the first 16 chars, mainly since the first few bits of the UUID are static: https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)